pkg/steps/template.go

```diff
@@ -620,6 +660,29 @@ func templateInstanceReady(instance *templateapi.TemplateInstance) (ready bool,
 	return false, nil
 }
 
+func podContainerNames(pod *coreapi.Pod) []string {
+	var names []string
+	for _, status := range append(append([]coreapi.ContainerStatus{}, pod.Status.InitContainerStatuses...), pod.Status.ContainerStatuses...) {
+		names = append(names, status.Name)
+	}
+	sort.Strings(names)
+	return names
+}
+
+func podTeardownCompleted(pod *coreapi.Pod) bool {
```

Comment:
this technically returns `true` when there is no `teardown` container at all ...

Reply:
Right, if there is no teardown container there is no need to additionally wait for it to complete